addons/mrp_subcontracting_purchase/models/account_move_line.py

```diff
@@ -14,3 +14,24 @@ def _get_price_unit_val_dif_and_relevant_qty(self):
             components_cost -= sum(subcontract_production.move_raw_ids.stock_valuation_layer_ids.mapped('value'))
             price_unit_val_dif = price_unit_val_dif + components_cost / relevant_qty
         return price_unit_val_dif, relevant_qty
+
+    def _get_valued_in_moves(self):
+        res = super()._get_valued_in_moves()
+        # subcontracted move valuations are not linked to the PO move but its orig move (the MO finished move)
+        res |= res.filtered(lambda m: m.is_subcontract).move_orig_ids
+        return res
+
+    def _get_layer_price_unit(self, layer):
+        """ For a subcontracted product, the AML is only concerned with the subcontracting cost (the price on the PO)
+            This override deducts the value of subcomponents from the layer price.
+        """
+        components_price = 0
+        production = layer.stock_move_id.production_id
+        if production.subcontractor_id and production.state == 'done':
+            # each layer has a quantity and price for each move, to get the correct component price for each move
+            # we need to get the components used for each quantity
+            for move in production.move_raw_ids:
+                component_qty_per_final_product_unit = move.product_qty / production.product_uom_qty
+                _qty_valued, qty_value = move.stock_valuation_layer_ids._consume_specific_qty(0, component_qty_per_final_product_unit)
+                components_price += qty_value
```

Comment:
why not simply `move.stock_valuation_layer_ids.value` ? like in 
https://github.com/odoo/odoo/blob/2926757ff67087d172bdcfdeccb133b4024b4081/addons/mrp_account/models/mrp_production.py#L94